### DISCUSSION TITLE
Cannot push docker image into docker hub from Github Action

### DISCUSSION THREAD

**Discussion by MohdRashid01** [community/community#24693] [Apps, API and Webhooks]
<p>Hi All,</p>
<p>I have yaml file in GitHub action and i have successfully build docker image in it and i want to push into docker hub but getting below error</p>
<pre><code class="lang-auto">Run docker push ***/vampi_docker:latest
docker push ***/vampi_docker:latest
shell: /usr/bin/bash -e {0}
An image does not exist locally with the tag: ***/vampi_docker
The push refers to repository [docker.io/***/vampi_docker]
Error: Process completed with exit code 1.
</code></pre>
<p>Here is the yml file</p>
<pre><code class="lang-auto">name: vampi_docker
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: docker login
      env:
        DOCKER_USER: ${{secrets.DOCKER_USER}}
        DOCKER_PASSWORD: ${{secrets.DOCKER_PASSWORD}}
        repository: rashidmd/vampi_docker:latest
        tags: latest, ${{ secrets.DOCKER_TOKEN }}
      run: |
        docker login -u $DOCKER_USER -p $DOCKER_PASSWORD
        
    - name: Build the Vampi Docker image
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt
        docker build . --file Dockerfile --tag vampi_docker:latest
        
    - name: List images
      run: docker images    
    - name: Docker Push
      run: docker push ${{secrets.DOCKER_USER}}/vampi_docker:latest
</code></pre>
<p>Please do let me know where im wrong and what i miss</p>

**Comment by airtower-luna** [comment]
<p>You need to tag the newly built image with the full Docker Hub name before pushing, e.g.:</p>
<pre><code class="lang-auto">docker tag vampi_docker:latest ${{secrets.DOCKER_USER}}/vampi_docker:latest
</code></pre>

**Comment by MohdRashid01** [comment]
<p>i have added like this</p>
<pre><code class="lang-auto"> name: Docker Push
     # run: docker push ${{secrets.DOCKER_USER}}/vampi_docker:latest
        run: docker tag vampi_docker:latest ${{secrets.DOCKER_USER}}/vampi_docker:latest
</code></pre>
<p>Job executed successfully but image doesn’t push into docker hub …i check docker hub it showing empty</p>

**Comment by airtower-luna** [comment]
<p>You need to do both. Tag first, then push.</p>

**Comment by MohdRashid01** [comment]
<p>problem solve but i want to deploy container using that image in github action can u tell me how to do it …any links or steps will be great for me in doing</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="MohdRashid01" data-post="5" data-topic="203826">
<div class="title">
<div class="quote-controls"></div>
 MohdRashid01:</div>
<blockquote>
<p>i want to deploy container using that image in github action</p>
</blockquote>
</aside>
<p>What exactly you want to do and achieve?</p>

**Comment by MohdRashid01** [comment]
<p>i want to run that image as a docker container in github action …i already have deploy image in github action now want to run that image as container …thats my task</p>

**Comment by airtower-luna** [comment]
<p>If you just want to run the container, you can do a <code>docker run</code> command as you would locally.</p>
<p>If you want to keep the container running after the Actions workflow completes you need to do that elsewhere (e.g. with a hoster), how to start it there from Actions depends on where the “elsewhere” is.</p>

**Comment by MohdRashid01** [comment]
<p>Yes, i want my container to be running please give us any link or website or any command to do it…hoster i didnt understand it …if u r saying about VM deploy that also i can do it…just need commands or website</p>

**Comment by juliaaano** [comment, marked as answer]
<p>Consider using an action to build and push the image. It will be less error prone, it handles the login and any other post actions (logout). I find it easier to maitain:</p>
<pre><code class="lang-auto">- name: build image
  id: build-image
  uses: redhat-actions/buildah-build@v2
  with:
    image: myuser/myimage
    tags: latest
    dockerfiles: |
      ./Dockerfile

- name: push image to ghcr.io
  uses: redhat-actions/push-to-registry@v2
  with:
    image: ${{ steps.build-image.outputs.image }}
    tags: ${{ steps.build-image.outputs.tags }}
    registry: ghcr.io
    username: ${{ github.actor }}
    password: ${{ secrets.GITHUB_TOKEN }}
</code></pre>

**Comment by DB11-Deep** [comment]
I have my "Dockerfile" present on the same branch where github actions is there. but still it is not able to locate the Dockerfile.

here is the code part:

      - name: build image
        id: build-image
        uses: redhat-actions/buildah-build@v2
        with:
          image: myorg/myimage
          tags: latest
          base-image: ubuntu:latest
          dockerfiles: |
            ./Dockerfile

my Dockerfile is present in the same branch along with otherfiles. please suggest with the update code to consider correct dockerfile.
